hail/src/main/scala/is/hail/expr/ir/ExtractIntervalFilters.scala

```diff
@@ -267,10 +310,10 @@ object ExtractIntervalFilters {
             case _ => None
           }
         }
-      case Let(name, value, body) if name != es.rowRef.name =>
+      case Let(name, value, body) =>
         // TODO: thread key identity through values, since this will break when CSE arrives
         // TODO: thread predicates in `value` through `body` as a ref
```

Comment:
What are these TODOs about? What would it mean to do them?

Reply:
Thanks, I forgot to update the todos.

The problem with the old `ExtractIntervalFilters` was that, like many of our analyses and optimizations, we couldn't see through Let nodes. This manifested in two ways, corresponding to the two comments:

1. We only recognized a key field when it was literaly `GetField(Ref("row"), keyField)`. If it was `Let(row, Ref("row"), ... GetField(row, keyField) ...)`, we didn't recognize it as a key field, and so wouldn't extract filters on the field to intervals of the key type.
2. If `value` is a predicate, we didn't try to extract intervals from it. For example, if the filter predicate is `Let(p, keyField < 10, ... p ... p)`, we didn't even try to lift the `keyField < 10` part. If `p` were inlined, however, we would analyze it correctly.

This PR addresses the first, but not the second. Over the weekend, I realized the right way to fix this, which addresses both restrictions, and I think fixes some other blind spots as well, and is a simpler design. I think the better change isn't much harder than this one was. I'm going to take a stab at it this morning, and make a PR if it doesn't take too long. But we can still merge this in the meantime since it fixes a blocking issue.